### PR TITLE
Remove deprecated code from MARC HTML view

### PR DESCRIPTION
**Pull request by hornc**
relates to #7959

The MARC HTML view feature still used deprecated code from `fast_parse.py` -- all test runs used to include the deprecation warnings.

This PR replaces the basic MARC view with the pymarc `as_dict()` representation of the `pymarc.Record`, and finally removes `fast_parse.py` entirely.

The tests are reduced to checking that an existing MARC8 encoded test data records is correctly converted to UTF-8. For the rest of the functionality, we'll trust pymarc.

**Update:** This also removes the deprecated module `marc_subject.py`.

### Screenshot

Before:

Screenshot from 2023-06-13 13-44-05

After:

Screenshot from 2023-06-13 16-14-52

### CODE REVIEW

File: openlibrary/catalog/marc/tests/test_marc_html.py [hunk 1]
```diff
@@ -1,46 +1,15 @@
+import os
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_html.py:1]: Would be cooler to see `from pathlib import Path` but that is a developer choice.  ;-)

**Review by cclauss — APPROVED** [review]
Looks good to me.

File: openlibrary/catalog/marc/tests/test_get_subjects.py [hunk 2]
```diff
@@ -235,13 +235,13 @@
 ]
 
 record_tag = '{http://www.loc.gov/MARC21/slim}record'
-
+TEST_DATA = f'{Path(__file__).parent}/test_data'
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_get_subjects.py:238]:
```suggestion
TEST_DATA = Path(__file__).parent / 'test_data'
```

File: openlibrary/catalog/marc/tests/test_get_subjects.py [hunk 3]
```diff
@@ -235,13 +235,13 @@
...
 
 class TestSubjects:
     @pytest.mark.parametrize('item,expected', xml_samples)
     def test_subjects_xml(self, item, expected):
         filename = (
-            os.path.dirname(__file__) + '/test_data/xml_input/' + item + '_marc.xml'
+            f'{TEST_DATA}/xml_input/{item}_marc.xml'
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_get_subjects.py]:
```suggestion
            str(TEST_DATA / 'xml_input' / f'{item}_marc.xml')
```

File: openlibrary/catalog/marc/tests/test_get_subjects.py [hunk 4]
```diff
@@ -251,7 +251,7 @@ def test_subjects_xml(self, item, expected):
 
     @pytest.mark.parametrize('item,expected', bin_samples)
     def test_subjects_bin(self, item, expected):
-        filename = os.path.dirname(__file__) + '/test_data/bin_input/' + item
+        filename = f'{TEST_DATA}/bin_input/{item}'
         with open(filename, mode='rb') as f:
             rec = MarcBinary(f.read())
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_get_subjects.py:255]:
```suggestion
        rec = MarcBinary((TEST_DATA / 'bin_input' / item).read_bytes())
```

File: openlibrary/catalog/marc/tests/test_marc_binary.py [hunk 5]
```diff
@@ -1,8 +1,7 @@
-import os
-
+from pathlib import Path
 from openlibrary.catalog.marc.marc_binary import BinaryDataField, MarcBinary
 
-test_data = "%s/test_data/bin_input/" % os.path.dirname(__file__)
+TEST_DATA = f'{Path(__file__).parent}/test_data/bin_input/'
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_binary.py:4]:
```suggestion
TEST_DATA = Path(__file__).parent / 'test_data' / 'bin_input'
```

File: openlibrary/catalog/marc/tests/test_marc_binary.py [hunk 6]
```diff
@@ -17,7 +16,7 @@ def marc8(self):
 
 
 def test_wrapped_lines():
-    filename = '%s/wrapped_lines.mrc' % test_data
+    filename = f'{TEST_DATA}/wrapped_lines.mrc'
     with open(filename, 'rb') as f:
         rec = MarcBinary(f.read())
         ret = list(rec.read_fields(['520']))
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_binary.py:22]:
```suggestion
    filepath = TEST_DATA / 'wrapped_lines.mrc'
    rec = MarcBinary(filepath.read_bytes())
    ret = list(rec.read_fields(['520']))
```

File: openlibrary/catalog/marc/tests/test_marc_binary.py [hunk 7]
```diff
@@ -49,7 +48,7 @@ def test_bad_marc_line(self):
 
 class Test_MarcBinary:
     def test_read_fields_returns_all(self):
-        filename = f'{test_data}/onquietcomedyint00brid_meta.mrc'
+        filename = f'{TEST_DATA}/onquietcomedyint00brid_meta.mrc'
         with open(filename, 'rb') as f:
             rec = MarcBinary(f.read())
             fields = list(rec.read_fields())
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_binary.py:54]:
```suggestion
        filepath = TEST_DATA / 'onquietcomedyint00brid_meta.mrc'
        rec = MarcBinary(filepath.read_bytes())
        fields = list(rec.read_fields())
```

File: openlibrary/catalog/marc/tests/test_marc_binary.py [hunk 8]
```diff
@@ -67,7 +66,7 @@ def test_read_fields_returns_all(self):
             assert isinstance(f100, BinaryDataField)
 
     def test_get_subfield_value(self):
-        filename = '%s/onquietcomedyint00brid_meta.mrc' % test_data
+        filename = f'{TEST_DATA}/onquietcomedyint00brid_meta.mrc'
         with open(filename, 'rb') as f:
             rec = MarcBinary(f.read())
             author_field = rec.get_fields('100')
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_binary.py:72]:
```suggestion
        filepath = TEST_DATA / 'onquietcomedyint00brid_meta.mrc'
        rec = MarcBinary(filepath.read_bytes())
        author_field = rec.get_fields('100')
```

File: openlibrary/catalog/marc/tests/test_marc_html.py [hunk 9]
```diff
@@ -1,8 +1,8 @@
-import os
+from pathlib import Path
 from openlibrary.catalog.marc.html import html_record
 
 
-TEST_DATA = f'{os.path.dirname(__file__)}/test_data/bin_input/'
+TEST_DATA = f'{Path(__file__).parent}/test_data/bin_input/'
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_marc_html.py:5]:
```suggestion
TEST_DATA = Path(__file__).parent / 'test_data' / 'bin_input'
```
Other modifications are required below...

**Review by cclauss — CHANGES_REQUESTED** [review]
https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_bytes

File: openlibrary/catalog/marc/tests/test_parse.py [hunk 10]
```diff
@@ -114,12 +114,12 @@ def test_xml(self, i):
 class TestParseMARCBinary:
     @pytest.mark.parametrize('i', bin_samples)
     def test_binary(self, i):
-        expect_filename = f'{test_data}/bin_expect/{i}'.replace('.mrc', '.json')
-        with open(f'{test_data}/bin_input/{i}', 'rb') as f:
+        expect_filename = f'{TEST_DATA}/bin_expect/{i}'.replace('.mrc', '.json')
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_parse.py:117]: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix

**Comment by hornc** [comment]
@cclauss Thanks for the review and suggestions, I've given all the MARC tests another pass focusing on using `Path` correctly.

Hopefully it is better now!

**Review by cclauss — APPROVED** [review]

**Pull request merged**